### PR TITLE
Move ownership of `AccessibilityBridgeWindows` to `FlutterWindowsView`

### PR DESCRIPTION
**Pull request by yaakovschectman**
Move ownership of the pointer to `AccessibilityBridgeWindows` from `FlutterWindowsEngine` to `FlutterWindowsView` in preparation for multi-window.

https://github.com/flutter/flutter/issues/124995

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/windows/flutter_windows_view.h [hunk 1]
```diff
@@ -200,10 +201,18 @@ class FlutterWindowsView : public WindowBindingHandlerDelegate,
   // |WindowBindingHandlerDelegate|
   virtual ui::AXFragmentRootDelegateWin* GetAxFragmentRootDelegate() override;
 
+  // Called to re/set the accessibility bridge pointer.
+  virtual void UpdateSemanticsEnabled(bool enabled);
```
Comment (yaakovschectman) [inline shell/platform/windows/flutter_windows_view.h:205]: The content of this new method could have instead been placed in `OnUpdateSemanticsEnabled`, which is what calls `FlutterWindowsEngine::UpdateSemanticsEnabled`. However, once we have multiple views, a call to `OnUpdateSemanticsEnabled` on a view would not necessarily call it for other views.

File: shell/platform/windows/flutter_windows_view.h [hunk 2]
```diff
@@ -200,10 +201,21 @@ class FlutterWindowsView : public WindowBindingHandlerDelegate,
...
+
+  std::shared_ptr<AccessibilityBridgeWindows> accessibility_bridge() {
```
Comment (loic-sharma) [inline shell/platform/windows/flutter_windows_view.h:207]: Should this be a weak_ptr?

**Review by loic-sharma — APPROVED** [review]
Looks good!

File: shell/platform/windows/flutter_windows_engine.cc [hunk 3]
```diff
@@ -345,16 +345,17 @@ bool FlutterWindowsEngine::Run(std::string_view entrypoint) {
 
     for (size_t i = 0; i < update->node_count; i++) {
       const FlutterSemanticsNode2* node = update->nodes[i];
-      host->accessibility_bridge_->AddFlutterSemanticsNodeUpdate(*node);
+      host->accessibility_bridge().lock()->AddFlutterSemanticsNodeUpdate(*node);
```
[the remark this answers is not in the document]
Comment (loic-sharma) [reply shell/platform/windows/flutter_windows_engine.cc:348]: This is a good idea - our plan is to create separate `FlutterSemanticsUpdate` for each view.

**Pull request merged**